apis/python/src/tiledbsoma/_sparse_nd_array.py

```diff
@@ -442,4 +464,134 @@ def tables(self) -> TableReadIter:
         Lifecycle:
             Experimental.
         """
+        self.array._set_reader_coords(self.sr, self.coords)
         return TableReadIter(self.sr)
+
+    @overload
+    def scipy(
+        self,
+        axis: Literal[0, 1] = 0,
+        step: Optional[int] = None,
+        compress: Literal[False] = False,
+        reindex_sparse_axis: bool = True,
+    ) -> Iterator[
+        Tuple[
+            Tuple[npt.NDArray[np.int64], npt.NDArray[np.int64]], scipy.sparse.coo_matrix
+        ]
+    ]:
+        ...
+
+    @overload
+    def scipy(
+        self,
+        axis: Literal[0],
+        step: Optional[int] = None,
+        compress: Literal[True] = True,
+        reindex_sparse_axis: bool = True,
+    ) -> Iterator[
+        Tuple[
+            Tuple[npt.NDArray[np.int64], npt.NDArray[np.int64]], scipy.sparse.csr_matrix
+        ]
+    ]:
+        ...
+
+    @overload
+    def scipy(
+        self,
+        axis: Literal[1],
+        step: Optional[int] = None,
+        compress: bool = True,
+        reindex_sparse_axis: bool = True,
+    ) -> Iterator[
+        Tuple[
+            Tuple[npt.NDArray[np.int64], npt.NDArray[np.int64]], scipy.sparse.csc_matrix
+        ]
+    ]:
+        ...
+
+    def scipy(
+        self,
+        axis: Literal[0, 1] = 0,
+        step: Optional[int] = None,
+        compress: bool = True,
+        reindex_sparse_axis: bool = True,
+    ) -> Iterator[
+        Tuple[
+            Tuple[npt.NDArray[np.int64], npt.NDArray[np.int64]],
+            Union[
+                scipy.sparse.csr_matrix,
+                scipy.sparse.csc_matrix,
+                scipy.sparse.coo_matrix,
+            ],
+        ]
+    ]:
+        """
+        Returns an iterator of
+        `SciPy sparse matrix` <https://docs.scipy.org/doc/scipy/reference/sparse.html>
+        over a 2D SparseNDArray.
+
+        Args:
+            axis:
+                The axis (0 or 1) across which to step. If `compress` is True, the
+                axis also determines which dimension is compressed, which in turn
+                determines the return type. If compressed==True and axis==0, a CSR
+                matrix is returned. If compressed==True and axis==1, a CSC matrix
+                is returned.
+            step:
+                Number of rows or columns to return in each iterator step.
+            compress:
+                If True, a CSC or CSR matrix is returned, dependent on the value of the
+                `axis` argument. If False, a COO matrix is returned.
+            reindex_sparse_axis:
+                If False (default), the sparse axis will also be reindexed from soma_joinid
```

Comment:
The default value for `reindex_sparse_axis` looks to be True.

I think you got the True/False reversed?
"If True (default), the sparse axis will also be reindexed from soma_joinid to zero-based indices. If False, the sparse axis values will remain joinids."

Reply:
Yes, I inverted the condition in the docstring. Thanks for catching.

When `compress==True`:
* Axis being iterated over is the `compressed` axis
* Axis *not* being iterated over is the `sparse` axis
The reason for this nomenclature is that these formats only compress coordinates on one axis (rows for CSR, ...)

So:
* axis==0 implies compressed axis is obs/row
* axis==1 implies compressed axis is var/col